File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.cpp
class Solution {
public:
    bool uniformArray(vector<int>& A) {
        sort(A.begin(), A.end());

        int oddcnt = 0;
        int eventcnt = 0;

        for (int it: A) {
            if (it % 2 != 0) oddcnt++;
            else eventcnt++;
        }
        if (oddcnt == 0 || eventcnt == 0) return true;

        return A[0] % 2 != 0;
    }
};

File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.py
class Solution:
    def uniformArray(self, A: list[int]) -> bool:
        A.sort()
        n = len(A)
        x = A[0]
        if x % 2 == 0:
            parity = 'even'
        else:
            parity = 'odd'
        for i in range(1, n):
            t = A[i]
            if t % 2 == 0:
                p = 'evev'
            else: 
                p = 'odd'
            
            if parity != p:
                s = abs(t - x)
                if s % 2 == 0:
                    p = 'even'
                else: p = 'odd'

                if parity != p:
                    return False
        return True